src/query/ts/m3db/encoded_series_iterator.go

```diff
@@ -108,3 +122,77 @@ func (it *encodedSeriesIter) Close() {
 	// noop, as the resources at the step may still be in use;
 	// instead call Close() on the encodedBlock that generated this
 }
+
+func (it *encodedSeriesIter) Append(
+	iter encoding.SeriesIterator,
+	meta block.SeriesMeta,
+) {
+	it.seriesIters = append(it.seriesIters, iter)
+	it.seriesMeta = append(it.seriesMeta, meta)
+}
+
+// MultiSeriesIter returns batched series iterators for the block based on
+// given concurrency.
+func (b *encodedBlock) MultiSeriesIter(
+	concurrency int,
+) ([]block.SeriesIterBatch, error) {
+	fn := iteratorBatchingFn
```

Comment:
We are defaulting to this func in two places - here and in the `NewOptions` method. Might be better to only default in the `NewOptions` and then here either just return an error if the batching function isn't defined, or just return a single batch.

Reply:
Yeah this was mostly for dealing with tests, will update the tests instead